packages/frontend/src/components/accounts/recovery_setup/SetupRecoveryMethod.js

```diff
@@ -246,7 +245,35 @@ class SetupRecoveryMethod extends Component {
                     });
                 } catch (e) {
                     console.warn(e.code);
-                    await fundCreateAccount(accountId, recoveryKeyPair, method.kind);
+                    const isNewAccount = await checkIsNew(accountId);
+                    if (!isNewAccount) {
+                        const accessKeys = await wallet.getAccessKeys(accountId) || [];
+                        const publicKeys = accessKeys.map(key => key.public_key);
+                        const publicKeyExistsOnAccount = publicKeys.includes(recoveryKeyPair.publicKey.toString());
+
+                        if (publicKeyExistsOnAccount) {
+                            try {
+                                await wallet.saveAndMakeAccountActive(accountId);
+                                await wallet.addLocalKeyAndFinishSetup(accountId, method.kind, recoveryKeyPair.publicKey);
+                            } catch (e) {
+                                showCustomAlert({
+                                    success: false,
+                                    messageCodeHeader: 'error',
+                                    errorMessage: e.message
+                                });
+                                redirectTo('/recover-account');
+                            }
+                        } else {
+                            showCustomAlert({
```

Comment:
Do we need to re-throw from here to ensure that `this.setState({ settingUpNewAccount: false });` is called on L302, since we're not redirecting in this branch?